src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -190,32 +202,85 @@ export function createJourneyApi(
       const tripQuery = parseTripQueryString(input);
       if (!tripQuery) return;
 
-      const result = await client.query<
-        TripsWithDetailsQuery,
-        TripsWithDetailsQueryVariables
-      >({
-        query: TripsWithDetailsDocument,
-        variables: {
-          ...tripQuery.query,
-          arriveBy: false,
-          ...DEFAULT_JOURNEY_CONFIG,
-        },
-      });
+      let formatedResult;
+      let result;
+      if ('via' in tripQuery.query) {
+        result = await client.query<
+          ViaTripsWithDetailsQuery,
+          ViaTripsWithDetailsQueryVariables
+        >({
+          query: ViaTripsWithDetailsDocument,
+          variables: {
+            ...tripQuery.query,
+            ...DEFAULT_JOURNEY_CONFIG,
+          },
+        });
+
+        const { tripPatternCombinations, tripPatternsPerSegment } =
+          result.data.viaTrip;
+
+        // Create list to make all trip pattern combinations more accessable.
+        const oldTripPatternCombinationList = oldFindTripPatternCombinationList(
+          tripPatternCombinations,
+        );
+
+        // Find trip patterns from-via and via-to destination.
+        const tripPatternsFromVia = tripPatternsPerSegment[0].tripPatterns;
+        const tripPatternsViaTo = tripPatternsPerSegment[1].tripPatterns;
+
+        // Find all possible trip patterns where the legs from the from-via location and the via-to location are concatinated.
+        const tripPatternsFromViaTo = generateTripPatternsFromViaToWithDetails(
+          oldTripPatternCombinationList,
+          tripPatternsFromVia,
+          tripPatternsViaTo,
+        );
+
+        const tripPatternCombinationList = findTripPatternCombinationList(
+          tripPatternCombinations,
+        );
+
+        let tripPatternIndexes = tripPatternCombinationList[0];
+        if (tripPatternCombinationList.length > 1) {
+          for (let i = 0; i < tripPatternCombinationList.length - 1; i++) {
+            tripPatternIndexes = findCombinedTripPatternsFromViaQuery(
+              tripPatternIndexes,
+              tripPatternCombinationList[i + 1],
+            );
+          }
+        }
+        const tripPatternsFromViaTo2 =
+          generateTripPatternsWithViaLoactionsWithDetails(
+            tripPatternIndexes,
+            tripPatternsPerSegment,
+          );
 
-      if (result.error || result.errors) {
-        throw result.error || result.errors;
+        formatedResult = { trip: tripPatternsFromViaTo2 };
+      } else {
+        result = await client.query<
+          TripsWithDetailsQuery,
+          TripsWithDetailsQueryVariables
+        >({
+          query: TripsWithDetailsDocument,
+          variables: {
+            ...tripQuery.query,
+            arriveBy: false,
+            ...DEFAULT_JOURNEY_CONFIG,
+          },
+        });
+
+        formatedResult = { trip: result.data.trip.tripPatterns };
       }
 
+      if (result.error || result.errors) throw result.error || result.errors;
```

Comment:
I think we should do this check earlier, right? In case of an error we don't have data to format

Reply:
I´ll ensure that the check is performed after receiving the result👍